engine/crates/engine/src/response/mod.rs

```diff
@@ -38,8 +45,33 @@ pub struct Response {
     #[serde(skip)]
     pub http_headers: HeaderMap,
 
-    /// GraphQL operation type derived from incoming request
-    pub operation_type: OperationType,
+    /// GraphQL operation.
+    pub graphql_operation: Option<ResponseOperation>,
```

Comment:
Not really a comment on your change since we were doing it already, but feels kind of bad passing data back in the response like this....

Reply:
We actually `take()` it in the gateway in my API PR to hide it from the client, but you're right – I should add `#[serde(skip_serializing_if = "Option::is_none")]`